pkg/client/unit.go

```diff
@@ -181,7 +181,9 @@ func (u *Unit) UpdateState(state UnitState, message string, payload map[string]i
 		u.stateMsg = message
 		changed = true
 	}
-	if (u.statePayload == nil && statePayload != nil) || (u.statePayload != nil && statePayload == nil) || !reflect.DeepEqual(u.statePayload, statePayload) {
+	if (u.statePayload == nil && statePayload != nil) ||
```

Comment:
nit: the reformatting of code you didn't actually change is making this review harder to do :)